lib/SILOptimizer/Analysis/EscapeAnalysis.cpp

```diff
@@ -496,24 +490,19 @@ void EscapeAnalysis::ConnectionGraph::initializePointsTo(CGNode *initialNode,
       pointsToEdges.push_back(node);
     }
     updatedNodes.push(node);
-    if (!isSuccessor)
-      backwardReachable.insert(node);
-
     return true;
   };
+  // Seed updatedNodes with initialNode.
+  visitDeferTarget(initialNode, true);
   // updatedNodes may grow during this loop.
-  unsigned nextUpdatedNodeIdx = 0;
-  for (; nextUpdatedNodeIdx < updatedNodes.size(); ++nextUpdatedNodeIdx)
-    updatedNodes[nextUpdatedNodeIdx]->visitDefers(visitDeferTarget);
+  for (unsigned idx = 0; idx < updatedNodes.size(); ++idx)
+    updatedNodes[idx]->visitDefers(visitDeferTarget);
   // Reset this worklist so others can be used, but updateNode.nodeVector still
   // holds all the nodes found by step 1.
   updatedNodes.reset();
 
   // Step 2: Update pointsTo fields by propagating backward from nodes that
   // already have a pointsTo edge.
-  assert(nextUpdatedNodeIdx == updatedNodes.size());
-  --nextUpdatedNodeIdx;
-  bool processBackwardReachable = false;
   do {
     while (!pointsToEdges.empty()) {
       CGNode *edgeNode = pointsToEdges.pop_back_val();
```

Comment:
Maybe it's worth mentioning in a comment here that edgeNode->setPointsToEdge is done either if
- the node a terminating node in the defer web, i.e. it has no defer-successors, or
- the node is in a terminating defer-edge-cycle

Reply:
Added a comment
```
        // This node is either (1) a leaf node in the defer web (identified in
        // step 1) or (2) an arbitrary node in a defer-cycle (identified in a
        // previous iteration of the outer loop).
```